internal/response-consumer/handler.go

```diff
@@ -255,16 +255,69 @@ func inferStatus(events *[]message.PlaybookRunResponseMessageYamlEventsElem, hos
 	}
 }
 
-func inferSatStatus(events *[]message.PlaybookSatRunResponseMessageYamlEventsElem, host *string) string {
+func satStatusEventDbMap(status message.PlaybookSatRunResponseMessageYamlEventsElemStatus) string {
+	switch {
+	case status == EventSatStatusSuccess:
+		return db.RunStatusSuccess
+	case status == EventSatStatusFailure:
+		return db.RunStatusFailure
+	case status == EventSatStatusCanceled:
+		return db.RunStatusCanceled
+	default:
+		return db.RunStatusRunning
+	}
+}
+
+func inferSatPlaybookStatus(events *[]message.PlaybookSatRunResponseMessageYamlEventsElem) string {
+	hostStatusMap := make(map[string]string)
+
+	for _, event := range *events {
+		if event.Type == EventSatPlaybookCompleted {
+			return satStatusEventDbMap(*event.Status)
+		}
+
+		if event.Host != nil {
+			if _, ok := hostStatusMap[*event.Host]; !ok {
+				hostStatusMap[*event.Host] = db.RunStatusRunning
+			}
+			if event.Status != nil {
+				hostStatusMap[*event.Host] = satStatusEventDbMap(*event.Status)
+			}
+		}
+	}
+
+	failed := false
+	canceled := false
+
+	for _, status := range hostStatusMap {
+		if status == db.RunStatusRunning {
+			return status
+		}
+		if status == db.RunStatusFailure {
+			failed = true
+		}
+		if status == db.RunStatusCanceled {
+			canceled = true
+		}
+	}
+
+	switch {
+	case failed && canceled || failed:
+		return db.RunStatusFailure
+	case canceled:
+		return db.RunStatusCanceled
+	default:
+		return db.RunStatusSuccess
+	}
+}
+
+func inferSatHostStatus(events *[]message.PlaybookSatRunResponseMessageYamlEventsElem, host string) string {
```

Comment:
Now that the logic for infering status of host/playbook has been split into separate functions (which is great), this function can be significantly simplified.